### PR TITLE
fix(db): do custom migrations for releases iteratively

### PR DESCRIPTION
**Pull request by AminSlk**
Ref: SRX-0Q0FWU

### CODE REVIEW

**Comment by jonaswittbrodt** [comment]
You need to move the transaction inside the for loop.

File: pkg/db/db.go [hunk 1]
```diff
@@ -5467,14 +5468,19 @@ func (h *DBHandler) RunCustomMigrationReleaseEnvironments(ctx context.Context) e
 	span, ctx := tracer.StartSpanFromContext(ctx, "RunCustomMigrationReleaseEnvironments")
 	defer span.Finish()
 	return h.WithTransaction(ctx, false, func(ctx context.Context, transaction *sql.Tx) error {
```
Comment (sven-urbanski-freiheit-com) [inline pkg/db/db.go:5470]: with transaction must be called more than once

File: pkg/db/db.go [hunk 2]
```diff
@@ -5467,14 +5468,19 @@ func (h *DBHandler) RunCustomMigrationReleaseEnvironments(ctx context.Context) e
...
+		for {
+			releasesWithoutEnvironments, err := h.DBSelectReleasesWithoutEnvironments(ctx, transaction)
+			if len(releasesWithoutEnvironments) == 0 {
+				break
+			}
 			if err != nil {
-				return fmt.Errorf("could not insert release, error: %w", err)
+				return fmt.Errorf("could not get releases without environments, error: %w", err)
+			}
+			for _, release := range releasesWithoutEnvironments {
```
Comment (sven-urbanski-freiheit-com) [inline pkg/db/db.go:5479]:
```suggestion
                        logger.FromContext(ctx).Sugar().Infof("deleting %d releases", len(releasesWithoutEnvironments))
			for _, release := range releasesWithoutEnvironments {
```

File: pkg/db/db.go (hunk 1 above)
Comment (AminSlk) [reply pkg/db/db.go]: done

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**